### PR TITLE
Porting Android doc from Google Docs

### PR DESCRIPTION
**Pull request by secondsun**
Port of a previous Android SDK document from Google Drive

### CODE REVIEW

**Comment by wtrocki** [comment]
Proposals tend to have some specific structure and be clearly stating goal, motivation, set of steps etc. For SDK we will need to have first top level strategy around repositories etc. 
This PR can be transformed to at least 5-8 different proposals, with lots of considerations inside. 

I fully agree with all points and options, but I think that we will need to tackle the most important elements that we need for the moment and then draft proposals for the rest at later stage. 
Proposals should be achievable in sprint. Going to draft mobile strategy in separate document and work to incorporate ideas from this PR.

File: android-sdk/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    * Setup auto publishing of SNAPSHOT builds
+
+* BOM style dependency management 
+    * Source a gradle plugin to support BOMs
+    * Ex [https://github.com/aerogear/aerogear-parent](https://github.com/aerogear/aerogear-parent)
+
+* Move everything to aerogear github organization
+  * What is included in everything?
+
+* Event driven API (either promise style or listeners. NO CALLBACKS! [They are hard to use in Android]
```
Comment (josemigallas) [inline android-sdk/README.md:19]: Aren't listeners just a kind of callback?

Comment (secondsun) [reply android-sdk/README.md:19]: Yes and no.  There is a bit more formality with listeners usually than callbacks which are usually written as anonymous inner classes inline with a function reference.  

The comment is basically saying that I want function calls to lean more towards formality than toward inline callbacks.  I can whip up some pseudo code if you like to demonstrate my point more clearly.

**Comment by secondsun** [comment]
closing per https://github.com/aerogear/proposals/pull/6#discussion_r159250764

**Pull request closed without merging**